shell/platform/android/io/flutter/embedding/android/FlutterImageView.java

```diff
@@ -0,0 +1,88 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.embedding.android;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.graphics.Canvas;
+import android.graphics.ColorSpace;
+import android.hardware.HardwareBuffer;
+import android.media.Image;
+import android.media.Image.Plane;
+import android.media.ImageReader;
+import android.os.Build;
+import android.view.View;
+import androidx.annotation.Nullable;
+import androidx.annotation.RequiresApi;
+
+class FlutterImageView extends View implements RenderSurface {
+  private final ImageReader imageReader;
+  @Nullable private Image nextImage;
+  private Image currentImage;
+
+  public FlutterImageView(Context context, ImageReader imageReader) {
+    super(context, null);
+    this.imageReader = imageReader;
+  }
+
+  @RequiresApi(api = Build.VERSION_CODES.KITKAT)
+  public void acquireLatestImage() {
+    nextImage = imageReader.acquireLatestImage();
+    invalidate();
+  }
+
+  @RequiresApi(api = Build.VERSION_CODES.P)
+  @Override
+  protected void onDraw(Canvas canvas) {
+    super.onDraw(canvas);
+    if (nextImage == null) {
+      return;
+    }
+
+    currentImage.close();
+    currentImage = nextImage;
+    nextImage = null;
+
+    if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.P) {
+      drawImageBuffer(canvas);
+    } else if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
+      drawImagePlane(canvas);
+    }
```

Comment:
else: throw unsupported Android version

Reply:
Since, I moved the `@RequiresApi(api = Build.VERSION_CODES.KITKAT)` to the top of the class due to your other comment. I don't think it needs that anymore, right?